ktor-server/ktor-server-netty/jvm/src/io/ktor/server/netty/NettyApplicationCallHandler.kt

```diff
@@ -47,7 +50,30 @@ internal class NettyApplicationCallHandler(
         }
     }
 
-    public companion object {
+    companion object {
         private val CallHandlerCoroutineName = CoroutineName("call-handler")
     }
 }
+
+internal fun HttpRequest.isValid(): Boolean {
+    if (decoderResult().isFailure) {
+        return false
+    }
+
+    val encodings = headers().getAll(io.ktor.http.HttpHeaders.TransferEncoding) ?: return true
+
+    encodings.forEachIndexed { index, header ->
+        val chunkedStart = header.indexOf(CHUNKED_VALUE)
+
+        if (chunkedStart == -1) return@forEachIndexed
+
+        if (index + 1 != encodings.size) return false
+
+        val chunkedIsNotLast = chunkedStart + CHUNKED_VALUE.length < header.length
+        if (chunkedIsNotLast) {
+            return false
+        }
+    }
+
+    return true
```

Comment:
So looks like if we have some unsupported encodings, we accept it. Not sure about it, probably it could be fine so it will be possible to handle custom encodings in the app level, or we should fail right here with no chance. I am in doubt

Reply:
I think it's a different problem, let's discuss it separately